spark/v3.2/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkMetadataColumns.java

```diff
@@ -169,6 +170,35 @@ public void testPartitionMetadataColumnWithUnknownTransforms() {
         () -> sql("SELECT _partition FROM %s", TABLE_NAME));
   }
 
+  @Test
+  public void testConflictingColumns() {
+    table.updateSchema()
+        .addColumn(MetadataColumns.SPEC_ID.name(), Types.IntegerType.get())
+        .addColumn(MetadataColumns.FILE_PATH.name(), Types.StringType.get())
+        .commit();
```

Comment:
Eventually, would it make sense to fail here, in the `updateSchema` pathway?

Reply:
Maybe. I would like to avoid not allowing these names, but you're right that it would catch the problem earlier.